runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -306,11 +306,15 @@ end
 --- )
 --- ```
 ---
----@param _ lsp.ResponseError?
+---@param error lsp.ResponseError?
 ---@param result lsp.DocumentDiagnosticReport
 ---@param ctx lsp.HandlerContext
 ---@param config vim.diagnostic.Opts Configuration table (see |vim.diagnostic.config()|).
-function M.on_diagnostic(_, result, ctx, config)
+function M.on_diagnostic(error, result, ctx, config)
+  -- Ignore ServerCancelled error
+  if error ~= nil and error.code == -32802 then
```

Comment:
Instead of hardcoding the code here you should be adding it to protocol's error codes: https://github.com/neovim/neovim/blob/b4599acbf8e30aa1234bc4041546f960dcc3a238/runtime/lua/vim/lsp/protocol.lua#L163